cvl/internal/yparser/ly_path.go

```diff
@@ -105,28 +105,3 @@ type lyPathElem struct {
 // value contains both single and double quotes, resulting in a vague format.
 // Such predicates cannot be parsed.
 var lyPredicatePattern = regexp.MustCompile(`^\[([^=]*)=('[^']*'|"[^"]*")]`)
-
-// Regex patterns to extract target node name and value from libyang error message.
-// Example messages:
-// - Invalid value "9999999" in "vlanid" element
-// - Missing required element "vlanid" in "VLAN_LIST"
-// - Value "xyz" does not satisfy the constraint "Ethernet([1-3][0-9]{3}|[1-9][0-9]{2}|[1-9][0-9]|[0-9])" (range, length, or pattern)
-// - Leafref "/sonic-port:sonic-port/sonic-port:PORT/sonic-port:ifname" of value "Ethernet668" points to a non-existing leaf
-// - Failed to find "extra" as a sibling to "sonic-acl:aclname"
-var (
-	lyBadValue    = regexp.MustCompile(`[Vv]alue "([^"]*)" `)
```

Comment:
Is libyang3 providing any better way of finding the target node when there is a validation failure? This was also important for cvl error reporting. There was no such facility in libyang1.x and we had to grep the error message.

Reply:
Not really.  Get the table and key from the data xpath, and the value from the error message.  Basically same as v1 in that regard ... 

I'm still working on error mapping, but only as they come up in test cases ... right now I'm struggling to understand the leafref issues and why they worked in libyang1 but not libyang3 (put a comment on this ticket if you could look at it)

Reply:
I think we can safely remove this lyBadValue.

We want to return correct key & field names in the validation error object. In most of the cases libyang data xpath used to contain these case. But in some cases it was not providing the field name -- i guess add node cases. Hence those lyElemPrefix, lyElemSuffix and lyUnknownElem regex are used to extract the field name from the message itself. The lyBadValue is actually optional -- it tries extract the value create a format the message in "Field FF has invalid value VV" format. If you remove lyBadValue, the error message would be "Field FF has invalid value" which is also good enough IMO.

Reply:
If we remove the regex matching, then I've got to update a bunch of test cases.  I'm basically out of time here, its something that someone else can do later I'm sure if they want to.